sensor/tests/resource/role/role_test.go

```diff
@@ -69,6 +87,66 @@ func (s *RoleDependencySuite) Test_PermutationTest() {
 	)
 }
 
+func matchBinding(namespace, id string) resource.MatchResource {
+	return func(resource *central.MsgFromSensor) bool {
+		if resource.GetEvent() == nil || resource.GetEvent().GetBinding() == nil {
+			return false
+		}
+		return resource.GetEvent().GetBinding().GetId() == id && resource.GetEvent().GetBinding().GetNamespace() == namespace
+	}
+}
+
+func (s *RoleDependencySuite) Test_BindingHasNoRoleId() {
+	s.testContext.RunBare("Binding should get an update if role gets created after", func(t *testing.T, testC *resource.TestContext, _ map[string]k8s.Object) {
+		deleteDep, err := testC.ApplyFileNoObject(context.Background(), "sensor-integration", NginxDeployment)
+		defer utils.IgnoreError(deleteDep)
+		require.NoError(t, err)
+
+		var binding v12.RoleBinding
+		deleteRoleBinding, err := testC.ApplyFile(context.Background(), "sensor-integration", NginxRoleBinding, &binding)
+		defer utils.IgnoreError(deleteRoleBinding)
+		require.NoError(t, err)
+
+		testC.LastResourceState(matchBinding(binding.GetNamespace(), string(binding.GetUID())), assertBindingHasRoleId(""), "No RoleID")
+
+		var role v12.Role
+		deleteRole, err := testC.ApplyFile(context.Background(), "sensor-integration", NginxRole, &role)
+		defer utils.IgnoreError(deleteRole)
+		require.NoError(t, err)
+
+		testC.LastResourceState(matchBinding(binding.GetNamespace(), string(binding.GetUID())), assertBindingHasRoleId(string(role.GetUID())), "Has RoleID")
+
+		testC.GetFakeCentral().ClearReceivedBuffer()
+
+		utils.IgnoreError(deleteRole)
+		utils.IgnoreError(deleteRoleBinding)
+		utils.IgnoreError(deleteDep)
+
+		testC.GetFakeCentral().ClearReceivedBuffer()
```

Comment:
I think this block is redundant. Unless you want to make sure that the resources are deleted before calling `ClearReceivedBuffer`, in which case all the `defer utils.IgnoreError(deleteResource)` aren't needed.